ptf

```diff
@@ -544,7 +544,7 @@ def load_test_modules(config):
                 if modname in sys.modules:
                     mod = sys.modules[modname]
                 else:
-                    mod = imp.load_module(modname, *imp.find_module(modname, [root]))
+                    mod = importlib.import_module(modname)
```

Comment:
Last time I messed around with importlib it was a little bit more complicated than that. This is missing the root folder as path. 
I remember you had to use find_spec and load_module instead. 
```
def import_module(root_path, module_name):
    """ Try to import a module and class directly instead of the typical
        Python method. Allows for dynamic imports. """
    finder = importlib.machinery.PathFinder()
    # Unfortunately this does not support Posix paths and silently fails. So we have to use str().
    # This is a standard lib function...
    module_specs = finder.find_spec(str(module_name), [str(root_path)])
    return module_specs.loader.load_module()
```

Reply:
@fruffy is right here
The test failure means that no test specs can be collected, and this is definitely connected to this change.
You should be able to run the test scripts locally to reproduce.
Additionally this change means that Python 3.1 is required, so this line needs to be changed accordingly: https://github.com/p4lang/ptf/blob/9c1787d15a120a71735e4b39311c877418d90bb3/setup.cfg#L22 (note that I don't have an issue with this version requirement).

Reply:
I may have fixed this.  Still poking things with a stick in terms of my knowledge level :-)